apps/nextjs-website/src/app/auth/confirmation/page.tsx

```diff
@@ -0,0 +1,57 @@
+'use client';
+import { Box, CircularProgress, Stack } from '@mui/material';
+import { environment } from '@/config';
+import PageNotFound from '@/app/not-found';
+import { Auth } from 'aws-amplify';
+import { useSearchParams, useRouter } from 'next/navigation';
+import { useEffect, useState } from 'react';
+
+const Confirmation = () => {
+  const searchParams = useSearchParams();
+  const router = useRouter();
+  const username = searchParams.get('username');
+  const code = searchParams.get('code');
+
+  const [renderNotFound, setRenderNotFound] = useState(environment === 'prod');
+
+  useEffect(() => {
+    if (username && code) {
+      Auth.confirmSignUp(username, code)
+        .then(() => {
+          setRenderNotFound(false);
+          // eslint-disable-next-line functional/immutable-data
+          router.push('/auth/account-activated');
+        })
+        .catch(() => {
+          setRenderNotFound(true);
+        });
```

Comment:
What about to handle errors in a more specific way? If any error occurs the `configSignUp` endpoint returns a response like the following one:

```
{"name":"ExpiredCodeException","code":"ExpiredCodeException"}
```

You can find more details reading the [cognito docs](https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_ConfirmSignUp.html#API_ConfirmSignUp_Errors).

Choose yourself between create a ticket about a better error handling or do the improve now

Reply:
We add this [ticket](https://pagopa.atlassian.net/browse/DEV-1047) to handle mange login and signup errors